src/OpenTelemetry.Api/Trace/TracerProvider.cs

```diff
@@ -44,5 +51,92 @@ protected TracerProvider()
     /// <param name="version">Version of the instrumentation library.</param>
     /// <returns>Tracer instance.</returns>
     public Tracer GetTracer(string name, string? version = null)
-        => new(new ActivitySource(name ?? string.Empty, version));
+    {
+        var tracers = this.tracers;
+        if (tracers == null)
+        {
+            // Note: Returns a no-op Tracer once dispose has been called.
+            return new(activitySource: null);
+        }
+
+        var key = threadTracerKey == null
+            ? (threadTracerKey = new TracerKey(name, version))
+            : threadTracerKey.Update(name, version);
+
+        if (tracers[key] is not Tracer tracer)
+        {
+            lock (tracers)
+            {
+                tracer = (tracers[key] as Tracer)!;
+                if (tracer == null)
+                {
+                    if (this.tracers != null)
```

Comment:
```suggestion
                    if (tracers != null)
```

Reply:
This is intentional and key to the design! In `Dispose` the `Hashtable` is set to `null` **outside** of the lock. So this check here **inside** the lock is to detect that race with `Dispose`.

Reply:
PS: I updated the comment when I switched to `ConcurrentDictionary` so hopefully it is more clear.